dependencies.go

```diff
@@ -130,11 +153,13 @@ func (f *faultyFile) Write(p []byte) (int, error) {
 	f.d.mu.Lock()
 	defer f.d.mu.Unlock()
 
-	fail := fastrand.Intn(f.d.failDenominator) == 0
-	f.d.failDenominator++
-	if fail || f.d.failDenominator >= f.d.writeLimit {
-		f.d.failed = true
-		return len(p), nil
+	if !*f.d.disabled {
+		fail := fastrand.Intn(int(*f.d.failDenominator)) == 0
+		*f.d.failDenominator++
+		if fail || *f.d.failDenominator >= *f.d.writeLimit {
+			*f.d.failed = true
+			return len(p), nil
+		}
```

Comment:
I'm pretty sure you need to be checking `f.d.failed` at some point here, so that it fails if the disk has already failed.

Reply:
I would also like to see this function extended so that in the event of failure, it'll write random bytes to disk of the write call, but not all of them, which makes the corruption a lot more aggressive.

Reply:
Yes, you are right

Reply:
something like `return f.file.Write(fastrand.Bytes(fastrand.Intn(len(p) + 1)))` ?